packages/@atjson/renderer-react/src/index.ts

```diff
@@ -73,20 +58,22 @@ export default class ReactRenderer extends Renderer {
   ): Iterator<void, ReactElement | ReactElement[], ReactElement[]> {
     this.renderSubdocuments(annotation);
 
-    let AnnotationComponent =
-      this.componentLookup[annotation.type] ||
-      this.componentLookup[classify(annotation.type)];
+    const children = yield;
+    const key = `${annotation.id}-${annotation.start}`;
 
-    if (AnnotationComponent) {
-      return React.createElement(
-        AnnotationComponent,
-        annotation.attributes,
-        ...(yield)
-      );
-    } else {
-      // console.warn(`No component found for "${node.type}"- content will be yielded`);
-      let components = yield;
-      return components;
-    }
+    return React.createElement(ReactRendererConsumer, { key }, value => {
```

Comment:
Could we add some assertions here so folks are aware that they need to wrap the rendering in a context provider?